service-deployment/bootstrap/hadoop-service/hadoop-configuration/hdfs-start.py.template

```diff
@@ -0,0 +1,162 @@
+#!/usr/bin/env python
+
+# Copyright (c) Microsoft Corporation
+# All rights reserved.
+#
+# MIT License
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
+# documentation files (the "Software"), to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and
+# to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+# The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING
+# BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+# NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+# DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+import os
+import sys
+import kazoo
+import subprocess
+
+
+
+def execute_shell_with_output(shell_cmd, error_msg):
+
+    return_code = 0
+
+    try:
+        res = subprocess.check_output( shell_cmd, shell=True )
+
+    except subprocess.CalledProcessError as err:
+        print error_msg
+        return_code = err.returncode
+        res = err.output
+
+    return res, return_code
+
+
+
+def execute_shell(shell_cmd, error_msg):
+
+    try:
+        subprocess.check_call( shell_cmd, shell=True )
+
+    except subprocess.CalledProcessError:
+        print error_msg
+        sys.exit(1)
+
+
+
+if __name__ == "__main__":
+
+    # Will be filled when namenode bootstraping
+    host = "{{ host_config['ip'] }}"
+    isBootstrapping = False
+    hasActiveNameNode = False
+    isNewNameNode = os.listdir("/var/lib/hdfs/name") == []
+
+    # Will be filled when namenode bootstraping
+    zookeeper_quorum="
+{%- for host in cluster_config if 'zkid' in cluster_config[ host ] -%}
+{{cluster_config[ host ]['ip']}}:2181{% if not loop.last %},{% endif %}
+{%- endfor -%}
+"
```

Comment:
shall we increase the indent here?

Reply:
Better not. It's not a python syntax.

Following is a test result.
```
 host = "10.xx.xx.xx"
    isBootstrapping = False
    hasActiveNameNode = False
    isNewNameNode = os.listdir("/var/lib/hdfs/name") == []

    # Will be filled when namenode bootstraping
    zookeeper_quorum="10.xx.xx.xx:2181,10.xx.xx.xx:2181,10.xx.xx.xx:2181"
    zk = kazoo.client.KazooClient(hosts=zookeeper_quorum)
    zk.start()

```